src/main/java/io/sealights/plugins/sealightsjenkins/integration/upgrade/UpgradeProxy.java

```diff
@@ -53,9 +55,18 @@ public boolean downloadAgent(String urlToAgent, String destFile) throws IOExcept
         boolean isSuccess = true;
         logger.info("Trying to download agent from url '" + urlToAgent + "' to folder '" + destFile + "'.");
         try {
-            URL agentUrl = new URL(urlToAgent);
             File agentDestination = new File(destFile);
-            FileUtils.copyURLToFile(agentUrl, agentDestination);
+            ApacheHttpClient client = new ApacheHttpClient();
+            HttpResponse response = client.getFile(urlToAgent, upgradeConfiguration.getProxy(), upgradeConfiguration.getToken(), false);
+            InputStream responseStream = response.getResponseStream();
+            FileOutputStream agentFile = new FileOutputStream(new File(destFile));
+            int inByte;
+            while ((inByte = responseStream.read()) != -1) {
```

Comment:
Please check if there is a better way to do this (using buffer API instead of byte-by-byte)